From 4583c7177817c77dc9eef19aa0eec66b663515f3 Mon Sep 17 00:00:00 2001
From: Chaitanya Tata <Chaitanya.Tata@imgtec.com>
Date: Wed, 31 Aug 2016 21:49:08 +0530
Subject: [PATCH 3/3] uccp420wlan: Fix streamsync timestamps update in AP mode

As a part of checkpatch, the logic to update streamsync params was modified,
this introduced a bug where params are locally updated but not in the main
struct.

This broke streamsync feature in AP mode.

Change-Id: I74f8d311500410639bdb4b14f9ac9951a6b0f2b7
---
 drivers/net/wireless/uccp420wlan/src/tx.c | 20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/drivers/net/wireless/uccp420wlan/src/tx.c b/drivers/net/wireless/uccp420wlan/src/tx.c
index d281e23..c21e831 100755
--- a/drivers/net/wireless/uccp420wlan/src/tx.c
+++ b/drivers/net/wireless/uccp420wlan/src/tx.c
@@ -1125,19 +1125,19 @@ int uccp420wlan_tx_free_buff_req(struct mac80211_dev *dev,
 				unsigned int ldelta = 0;
 				int ets_band;
 				int bts_vif = uvif->vif_index;
-				struct wifi_sync sync =
-						dev->params->sync[bts_vif];
+				struct wifi_sync *sync =
+						&dev->params->sync[bts_vif];
 #ifdef MULTI_CHAN_SUPPORT
 				ets_band = get_band_chanctx(dev, uvif);
 #endif
 				spin_lock(&tsf_lock);
-				sync.status = 1;
-				ether_addr_copy(sync.bssid,
+				sync->status = 1;
+				ether_addr_copy(sync->bssid,
 					ivif->bss_conf.bssid);
-				memcpy(sync.ts1, tx_done->reserved, 8);
-				memcpy(&sync.ts2, (tx_done->reserved + 8), 4);
-				ts2 = sync.ts2;
-				sync.atu = 0;
+				memcpy(sync->ts1, tx_done->reserved, 8);
+				memcpy(&sync->ts2, (tx_done->reserved + 8), 4);
+				ts2 = sync->ts2;
+				sync->atu = 0;
 
 				if (ets_band == IEEE80211_BAND_2GHZ)
 					ldelta = BTS_AP_24GHZ_ETS;
@@ -1145,8 +1145,8 @@ int uccp420wlan_tx_free_buff_req(struct mac80211_dev *dev,
 					ldelta = BTS_AP_5GHZ_ETS;
 
 				if (frc_to_atu) {
-					frc_to_atu(ts2, &sync.atu, 0);
-					sync.atu += ldelta * 1000;
+					frc_to_atu(ts2, &sync->atu, 0);
+					sync->atu += ldelta * 1000;
 				}
 				spin_unlock(&tsf_lock);
 			}
-- 
2.6.2

